### PR TITLE
cleaning up old data-sparks page content

### LINKED ISSUE #2922: Remove the dbt-spark Adapter part of the `Connect your database` for Databricks
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

[page](https://docs.getdbt.com/docs/get-started/connect-your-database)

### What part(s) of the page would you like to see updated?

I would like the entire dbt-spark adapter section to be removed from the connect to Databricks. This is because we have the dbt-Databricks adapter and users should not be utilizing the dbt-spark adapter to connect to Databricks (given the improved experience and maintenance). 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by runleonarun**
Closes #2922 

* Removing [This old page](https://docs.getdbt.com/docs/cloud/connect-data-platform/connect-your-database) that contains outdated content that was reorganized last month. 
* Removing links to old age.
* Adding redirects for old content

You can refer to this [internal conversation](https://dbt-labs.slack.com/archives/C02NCQ9483C/p1677176515399219) for more context.

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Removing pages (delete if not applicable):
- [x ] was already removed from `website/sidebars.js`
- [x] Add an entry `_redirects`

### CODE REVIEW

File: _redirects [hunk 1]
```diff
@@ -137,7 +138,8 @@ docs/dbt-cloud/using-dbt-cloud/cloud-model-timing-tab /docs/deploy/dbt-cloud-job
 /docs/dbt-cloud/cloud-quickstart /docs/get-started/getting-started/set-up-dbt-cloud 301
 /docs/cloud-quickstart /docs/dbt-cloud/cloud-quickstart 301
 /docs/dbt-cloud/cloud-configuring-dbt-cloud /docs/get-started/getting-started/set-up-dbt-cloud 301
-/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database /docs/get-started/connect-your-database 301
+/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database /docs/cloud/connect-data-platform/about-connections 301
+/docs/get-started/connect-your-database /docs/cloud/connect-data-platform/about-connections 301
```
Comment (nghi-ly) [inline _redirects:142]: i don't think this line is needed -- you added it above on line 19

```suggestion

```

Comment (nghi-ly) [reply _redirects:142]: should [line 76 and 77](https://github.com/dbt-labs/docs.getdbt.com/blob/a49e28bad7be610d107710844691c26f381093b3/_redirects#L76-L77) also have redirects?

Comment (nghi-ly) [reply _redirects:142]: also, [lines 141 - 142](https://github.com/dbt-labs/docs.getdbt.com/blob/a49e28bad7be610d107710844691c26f381093b3/_redirects#L76-L77) look to be duplicates of your new lines in 19 and 20. sugg removing one of these sets

**Review by nghi-ly — COMMENTED** [review]
thank you! left a few comments in the redirects file

File: _redirects (hunk 1 above)
Comment (runleonarun) [reply _redirects:142]: Added redirects for 76/77 and moved line 20 down

File: website/docs/docs/cloud/develop-in-the-cloud.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ To develop in the Cloud IDE, make sure you have the following:
 
 - A [dbt Cloud account](https://cloud.getdbt.com/) and [Developer seat license](/docs/cloud/manage-access/seats-and-users)
 - A git repository set up and git provider must have `write` access enabled. See [Connecting your GitHub Account](/docs/cloud/git/connect-github) or [Importing a project by git URL](/docs/cloud/git/import-a-project-by-git-url) for detailed setup instructions
-- A dbt project connected to a [data platform](/docs/cloud/connect-data-platform/connect-your-database)
+- A dbt project connected to a [data platform](/docs/cloud/connect-data-platform/about-connections)
 - Set up a [development environment and development credentials](#access-the-cloud-ide)
```
Comment (runleonarun) [inline website/docs/docs/cloud/develop-in-the-cloud.md]:
```suggestion
- A [development environment and development credentials](#access-the-cloud-ide) set up
```

File: website/docs/docs/cloud/develop-in-the-cloud.md [hunk 3]
```diff
@@ -15,8 +15,8 @@ To develop in the Cloud IDE, make sure you have the following:
 
 - A [dbt Cloud account](https://cloud.getdbt.com/) and [Developer seat license](/docs/cloud/manage-access/seats-and-users)
 - A git repository set up and git provider must have `write` access enabled. See [Connecting your GitHub Account](/docs/cloud/git/connect-github) or [Importing a project by git URL](/docs/cloud/git/import-a-project-by-git-url) for detailed setup instructions
-- A dbt project connected to a [data platform](/docs/cloud/connect-data-platform/connect-your-database)
-- Set up a [development environment and development credentials](#access-the-cloud-ide)
+- A dbt project connected to a [data platform](/docs/cloud/connect-data-platform/about-connections)
+- A [development environment and development credentials](#access-the-cloud-ide) set up
```
Comment (runleonarun) [inline website/docs/docs/cloud/develop-in-the-cloud.md:19]: I made this change to trigger a new build.

**Review by nghi-ly — APPROVED** [review]
lgtm! ty! :shipit:

**Pull request merged**